test/int32.test.js

```diff
@@ -0,0 +1,512 @@
+    it('when vaglue is null', async function() {
+      const schema = new Schema({
+        int32: {
+          type: Schema.Types.Int32,
+          required: true
+        }
+      });
+      const Test = mongoose.model('Test', schema);
+
+      const doc = new Test({
+        int: null
+      });
+
+      const err = await doc.validate().then(() => null, err => err);
+      assert.ok(err);
+      assert.ok(err.errors['int32']);
+      assert.equal(err.errors['int32'].name, 'ValidatorError');
+      assert.equal(
+        err.errors['int32'].message,
+        'Path `int32` is required.'
+      );
+    });
+    it('when value is non-null', async function() {
+      const schema = new Schema({
+        int32: {
+          type: Schema.Types.Int32,
+          required: true
+        }
+      });
+      const Test = mongoose.model('Test', schema);
+
+      const doc = new Test({
+        int: 3
+      });
+
+      const err = await doc.validate().then(() => null, err => err);
+      assert.ok(err);
+      assert.ok(err.errors['int32']);
+      assert.equal(err.errors['int32'].name, 'ValidatorError');
+      assert.equal(
+        err.errors['int32'].message,
+        'Path `int32` is required.'
+      );
+    });
+  });
+
+  describe('special inputs', function() {
+    it('supports INT32_MIN as input', function() {
+      const schema = new Schema({
+        myInt: {
+          type: Schema.Types.Int32
+        }
+      });
+      const Test = mongoose.model('Test', schema);
+
+      const doc = new Test({
+        myInt: INT32_MIN
+      });
+      assert.strictEqual(doc.myInt, INT32_MIN);
+    });
+
+    it('supports INT32_MAX as input', function() {
+      const schema = new Schema({
+        myInt: {
+          type: Schema.Types.Int32
+        }
+      });
+      const Test = mongoose.model('Test', schema);
+
+      const doc = new Test({
+        myInt: INT32_MAX
+      });
+      assert.strictEqual(doc.myInt, INT32_MAX);
+    });
+
+    it('supports undefined as input', function() {
+      const schema = new Schema({
+        myInt: {
+          type: Schema.Types.Int32
+        }
+      });
+      const Test = mongoose.model('Test', schema);
+
+      const doc = new Test({
+        myInt: undefined
+      });
+      assert.strictEqual(doc.myInt, undefined);
+    });
+
+    it('supports null as input', function() {
+      const schema = new Schema({
+        myInt: {
+          type: Schema.Types.Int32
+        }
+      });
+      const Test = mongoose.model('Test', schema);
+
+      const doc = new Test({
+        myInt: null
+      });
+      assert.strictEqual(doc.myInt, null);
+    });
+  });
+
+  describe('valid casts', function() {
+    it('casts from string', function() {
+      const schema = new Schema({
+        myInt: {
+          type: Schema.Types.Int32
+        }
+      });
+      const Test = mongoose.model('Test', schema);
+
+      const doc = new Test({
+        myInt: -42
```

Comment:
If I'm reading this test correctly, myInt should be a string here

Reply:
Good catch!